### PR TITLE
perf: improve forwarded header and recovery middlewares

### PR DESCRIPTION
**Pull request by juliens**
### What does this PR do?
Improve forwarded header and recovery middlewares.

### Motivation
Better performance!

### More

~~- [ ] Added/updated tests~~
~~- [ ] Added/updated documentation~~

### Additional Notes
Co-authored-by: Ludovic Fernandez <ldez@users.noreply.github.com>

### CODE REVIEW

File: pkg/middlewares/forwardedheaders/forwarded_header.go [hunk 1]
```diff
@@ -125,50 +134,69 @@ func (x *XForwarded) rewrite(outreq *http.Request) {
...
 		}
 	}
 
 	x.rewrite(r)
 
 	x.next.ServeHTTP(w, r)
 }
+
+// unsafeHeader allows to manage Header values.
+// Must be used only when the header name are already a canonical key.
```
Comment (mpl) [inline pkg/middlewares/forwardedheaders/forwarded_header.go:186]: depending on what you want to mean, you need either "header names already are canonical keys" or "header name already is".

**Review by jbdoumenjou — APPROVED** [review]
LGTM :+1:

**Review by rtribotte — APPROVED** [review]
LGTM 👍

**Review by kevinpollet — APPROVED** [review]
LGTM 🚀

**Pull request merged**